Modules/_ctypes/stgdict.c

```diff
@@ -697,29 +697,38 @@ PyCStructUnionType_update_stgdict(PyObject *type, PyObject *fields, int isStruct
     stgdict->align = total_align;
     stgdict->length = len;      /* ADD ffi_ofs? */
 
-#define MAX_STRUCT_SIZE 16
+
+#if defined(__aarch64__) || defined(__arm__)
+#  define MAX_STRUCT_SIZE 32
+#else
+#  define MAX_STRUCT_SIZE 16
+#endif
 
     if (arrays_seen && (size <= MAX_STRUCT_SIZE)) {
         /*
-         * See bpo-22273. Arrays are normally treated as pointers, which is
-         * fine when an array name is being passed as parameter, but not when
-         * passing structures by value that contain arrays. On 64-bit Linux,
-         * small structures passed by value are passed in registers, and in
-         * order to do this, libffi needs to know the true type of the array
-         * members of structs. Treating them as pointers breaks things.
+         * See bpo-22273 and gh-110190. Arrays are normally treated as
+         * pointers, which is fine when an array name is being passed as
+         * parameter, but not when passing structures by value that contain
+         * arrays.
+         * On x86_64 Linux and Arm platforms, small structures passed by
```

Comment:
Arm platforms = aarch64 but also 32-bit like ARMv7?

Reply:
That's correct! In fact the macro #if above  includes __aarch64__ and __arm__. If you prefer I can be more explicit and add an extra comment above the macro.

Reply:
Since I'd prefer the other line changed, you might just as well make this one more explicit, too.

Reply:
I kept this the same but I've added a comment on top of the macro where I explain why we need the 32 bytes and what we mean by Arm platforms.